src/persist/examples/kafka_upsert.rs

```diff
@@ -49,8 +51,11 @@ fn construct_persistent_kafka_upsert_source<G: Scope<Timestamp = u64>>(
     // - TODO: This needs to be respectful of ts and diff
     // - TODO: Don't use read_to_end_flattened
     let mut prev_value_by_key = HashMap::new();
-    for ((k, v), _, _) in out_read.snapshot()?.read_to_end_flattened()? {
-        prev_value_by_key.insert(k, v);
+    for ((k, v), ts, diff) in out_read.snapshot()?.read_to_end_flattened()? {
+        if ts > start_ts {
```

Comment:
shouldn't this be `<` (or maybe `<=`)? to read the data "as of" a timestamp, you need the data below it, no?

Reply:
The construction is slightly unintuitive, but we're filtering out anything that is strictly above `start_ts` by `continue`'ing, so we should be taking into account all the data that is `<= start_ts`.

Reply:
🤦 carry on